framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/AClassWrapper.java

```diff
@@ -0,0 +1,169 @@
+     * Wraps the "fields" field of AClass.
+     */
+    private final Map<String, AFieldWrapper> fields = new HashMap<>();
+
+    /** The enum constants of the class, if it has any. */
+    private List<VariableElement> enumConstants;
+
+    /** The type element representing the class. */
+    private TypeElement typeElement = null;
+
+    /**
+     * Wrap an AClass.
+     *
+     * @param theClass the wrapped object
+     */
+    public AClassWrapper(AClass theClass) {
+        this.theClass = theClass;
+    }
+
+    /**
+     * Call before doing anything with a method. Interacts with scenelib and fetches or creates an
+     * appropriate AMethodWrapper object.
+     *
+     * <p>Results are interened.
+     *
+     * @param methodName the name of the method
+     * @return an AMethodWrapper representing the method
+     */
+    public AMethodWrapper vivifyMethod(String methodName) {
+        if (methods.containsKey(methodName)) {
+            return methods.get(methodName);
+        } else {
+            AMethodWrapper wrapper = new AMethodWrapper(theClass.methods.getVivify(methodName));
+            methods.put(methodName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the methods that have been vivified (that is, interacted with) on a class.
+     *
+     * @return a map from method name to the object representing the method
+     */
+    public Map<String, AMethodWrapper> getMethods() {
+        return ImmutableMap.copyOf(methods);
+    }
+
+    /**
+     * Call before doing anything with a field. Interacts with scenelib and fetches or creates an
+     * appropriate AFieldWrapper object.
+     *
+     * <p>Results are interned.
+     *
+     * @param fieldName the name of the field
+     * @param type the type of the field, which scenelib doesn't track
+     * @return an AField object representing the field
+     */
+    public AFieldWrapper vivifyField(String fieldName, TypeMirror type) {
+        if (fields.containsKey(fieldName)) {
+            return fields.get(fieldName);
+        } else {
+            AFieldWrapper wrapper =
+                    new AFieldWrapper(theClass.fields.getVivify(fieldName), type.toString());
+            fields.put(fieldName, wrapper);
+            return wrapper;
+        }
+    }
+
+    /**
+     * Get all the fields that have been vivified (that is, interacted with) on a class.
+     *
+     * @return a map from field name to the object representing the field
+     */
+    public Map<String, AFieldWrapper> getFields() {
+        return ImmutableMap.copyOf(fields);
+    }
+
+    /**
+     * Get the annotations on the class
+     *
+     * @return the annotations, directly from scenelib
+     */
+    public Collection<? extends Annotation> getAnnotations() {
+        return theClass.tlAnnotationsHere;
+    }
+
+    /**
+     * The type of the class
+     *
+     * @return a type element representing this class
```

Comment:
This might return null.  Say under what circumstances.  Or probably it should throw an exception if the field is null.